source/objects/TraceResult.cpp

```diff
@@ -59,7 +59,8 @@ TraceResult::TraceResult(
 	detailScale(mat.detailScale), detailBlendFactor(mat.detailBlendFactor),
 	detailBlendMode(mat.detailBlendMode), detailTint(mat.detailTint),
 	detailAlphaMaskBaseTexture(mat.detailAlphaMaskBaseTexture),
-	texScale(mat.texScale)
+	texScale(mat.texScale),
+	materialPath(&mat.path)
```

Comment:
This should copy to the `TraceResult` (The accel struct could be gc’d while the trace result is still in memory)

Reply:
In fact I'm going to look into how efficient my VTF copy constructor is, cause copying the textures in might also be doable now

Reply:
I see why you did this now, `std::string` isn't trivially destructible

Solution to this is just providing access to all common material properties through the `TraceResult`

Reply:
My idea didn't work, gonna have to use ptrs to strings cause Lua doesn't call `__gc` until far too late
I was copying IVTFTexture resources to Lua directly instead of having the user load them from a path

Reply:
![image](https://user-images.githubusercontent.com/49565664/191720920-e41fcd5d-a372-402b-ae69-c1d48e30cd1c.png)

Reply:
> My idea didn't work, gonna have to use ptrs to strings cause Lua doesn't call `__gc` until far too late I was copying IVTFTexture resources to Lua directly instead of having the user load them from a path

Actually I'm going to implement a `__gc` method for `TraceResults`